sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpMessageExtensions.cs

```diff
@@ -26,9 +27,21 @@ private static IEnumerable<Data> AsAmqpData(this IEnumerable<ReadOnlyMemory<byte
         {
             foreach (ReadOnlyMemory<byte> data in binaryData)
             {
+                ArraySegment<byte> segment;
+                if (!data.IsEmpty)
```

Comment:
Do we need this case?   If I'm [reading](https://source.dot.net/#System.Private.CoreLib/MemoryMarshal.cs,261) correctly, if we pass an empty `ReadOnlyMemory<T>` instance in, an empty array segment should be returned.

A quick LinqPad test seems to confirm:

```csharp
var x = new ReadOnlyMemory<byte>();
x.IsEmpty.Dump();  // This is true

if (MemoryMarshal.TryGetArray(x, out var y))
{
    y.Dump();  // this shows no items.
}
else
{
    throw new Exception("Unexpected result from marshalling");
}
```